gesture/gesture-control.py

```diff
@@ -109,11 +113,17 @@ def run(self):
 						vectorY = 0.0
 						vectorZ = 0.0
 						self.vectorLock.release()
-		 
+
 				# update the FPS counter
 				fps.update()
 				fps.stop()
-		 
+
+				finalTime = time.time()
+				# Append new data set after every quarter of a second
+				if writecsv and finalTime - initialTime > 0.25:
```

Comment:
Why every 1/4 of a second? What happens if the vector, for some reason, has not been updated in 1/4 of a second?

Why not append new dataset when there is new vectors to add?

Reply:
Also, even though time.time() returns a floating point, I doubt you will get the precision needed to compute `finalTime - initialTime > 0.25` (see [this document](https://www.tutorialspoint.com/python/time_time.htm))

Reply:
I was worried about the csv file getting to large, hence I set a duration until when the next set of values are appended. And I actually wasn't worried about the data not changing, what if the object we are tracking has not been moving? Don't we want to capture that as well?

Reply:
I doubt the CSV file will get too large as each vector would have a size of `3*sizeof(float) // 12 bytes` (approximately since I'm not 100% sure how Python would store the list). Besides, this current implementation only has to work for like 5 minutes; future implementations of this feature will cover all use cases.

Read the body of code below:

			# check to see if we are currently tracking an object
			if initBB is not None:
				# grab the new bounding box coordinates of the object
				(success, box) = self.tracker.update(frame)

				if success:
					(x, y, w, h) = [int(v) for v in box]
					cv2.rectangle(frame, (x, y), (x + w, y + h),
						(0, 255, 0), 2)

					vectorX += x + w/2 - preX
					vectorY += y + h/2 - preY
					vectorZ += 0.0 # Needs work
					preX = x + w/2
					preY = y + h/2

					# If unable to acquire the lock, don't update the vectors
					if self.vectorLock.acquire(False):
						self.vectorX = vectorX
						self.vectorY = vectorY
						self.vectorZ = vectorZ

						vectorX = 0.0
						vectorY = 0.0
						vectorZ = 0.0
						self.vectorLock.release()`

The vectors values are updated only if the tracker is successful in "tracking" the target, even if the target has not moved in position. Try it yourself with the following inserted code:


					# If unable to acquire the lock, don't update the vectors
					if self.vectorLock.acquire(False):
						self.vectorX = vectorX
						self.vectorY = vectorY
						self.vectorZ = vectorZ

						print(self.grabVector()) # New code

						vectorX = 0.0
						vectorY = 0.0
						vectorZ = 0.0
						self.vectorLock.release()`
`

Reply:
Ahh, now I see what you meant by "there is new vectors to add", basically when the object has been successfully tracked, the vectors will be updated (irrespective of whether it has moved or not).